src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -1165,25 +1186,34 @@ abstract class MergeIntoSuiteBase
 
   testAnalysisErrorsInExtendedMerge("delete condition - subquery")(
     mergeOn = "s.key = t.key",
-    delete(condition = "s.value in (select value from t)"))(
+    delete(condition = "s.newValue in (select oldValue from t)"))(
     errorStrs = Nil)  // subqueries fail for unresolved reference to `t`
 
   testAnalysisErrorsInExtendedMerge("insert condition - unknown reference")(
     mergeOn = "s.key = t.key",
-    insert(condition = "unknownAttrib > 1", values = "(key, value) VALUES (s.key, s.value)"))(
-    errorStrs = "INSERT condition" :: "unknownAttrib" :: Nil)
+    insert(condition = "unknownAttrib > 1", values = "(key, oldValue) VALUES (s.key, s.newValue)"))(
+    // Should show unknownAttrib as invalid ref and (key, newValue) as valid column names,
+    // but not show oldValue as a valid name as target columns cannot be present in insert clause.
+    errorStrs = "INSERT condition" :: "unknownAttrib" :: "key" :: "newValue" :: Nil,
+    notErrorStrs = "oldValue")
+
+  testAnalysisErrorsInExtendedMerge("insert condition - reference to target table column")(
```

Comment:
this test verifies that insert condition cannot have references to target columns.